app/controllers/two_factor_authentication/otp_verification_controller.rb

```diff
@@ -23,21 +23,27 @@ def show
     def create
       result = otp_verification_form.submit
       post_analytics(result)
+
+      if !UserSessionContext.confirmation_context?(context)
+        handle_verification_for_authentication_context(
```

Comment:
positive check maybe easier to read, and matches the `_for_authentication_context` method inside?

```suggestion
      if UserSessionContext.authentication_or_reauthentication_context?(context)
        handle_verification_for_authentication_context(
```

Reply:
Ooh yeah I like that better 👍